### ISSUE TITLE
Avoid checks on equality of float results, use almostEqual instead

### ISSUE THREAD

**Issue by jakimowb** [EnMAP-Box/enmap-box#813]
Depending on the platform where they are calculated, operations that return floating point numbers may have different outputs.
The comparison of float outputs against all decimal places of a static float therefore can easily fail on different platforms   
Therefore the following test fails on macOS: 

For example `tests/enmap-box/enmapboxprocessing/test_utils.py` fails on macOS / macBook Pro 2020 (1,4 GHz Quad-Core Intel Core i5):
```
    def test_transformExtent(self):
        # different CRS
        extent = QgsRasterLayer(enmap).extent()
        crs = QgsRasterLayer(enmap).crs()
        toCrs = QgsCoordinateReferenceSystem().fromEpsgId(4326)
        self.assertEqual(
            QgsRectangle(13.24539916899924208, 52.41260765598481441, 13.34667529330139502, 52.52184188795437336),
            Utils.transformExtent(extent, crs, toCrs)
        )

```
Failure:
````bash
FAILED tests/enmap-box/enmapboxprocessing/test_utils.py::TestUtils::test_transformExtent - AssertionError: <QgsRectangle: 13.24539916899924208 52.41260765598481441, 13.34667529330139502 52.52184188795437336> != <QgsRectangle: 13.24539916899924208 52.41260765598481441, 13.34667529330139502 52.52184188795436626>
````

# Solution:
Compare equality of floats only on n-first decimal places, similar to https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual

**Comment by jakimowb** [comment]
@janzandr https://github.com/EnMAP-Box/enmap-box/commit/6a16af5bc1c72734251263544b2d6c5745886976 adds a workaround that uses `assertGeometriesEqual()`, available since QGIS 3.34. 

Feel free to adjust, but I think for coordinate values, a precision of 14 is more than ok
https://www.explainxkcd.com/wiki/index.php/2170:_Coordinate_Precision

**Comment by jakimowb** [comment]
fixed with #814

**Comment by jakimowb** [comment]
@janzandr feel free to modify the changes
